nucliadb/nucliadb/writer/api/v1/resource.py

```diff
@@ -317,10 +367,51 @@ async def delete_resource(
 )
 @requires(NucliaDBRoles.WRITER)
 @version(1)
-async def reindex_resource(request: Request, kbid: str, rid: str):
+async def reindex_resource(
+    request: Request,
+    kbid: str,
+    rid: Optional[str] = None,
+    rslug: Optional[str] = None,
+):
+
+    rid = await get_rid_from_params_or_raise_error(kbid, rid, rslug)
```

Comment:
Just as a comment, when you redefine the rid from Optional[str] to str its better that you create a different var name that type checker knows it may not be empty